TreeTracker/UI/Views/View Trees/ViewTreesViewController.swift

```diff
@@ -0,0 +1,59 @@
+//
+//  ViewTreesViewController.swift
+//  TreeTracker
+//
+//  Created by Remi Varghese on 10/27/20.
+//  Copyright © 2020 Greenstand. All rights reserved.
+//
+
+import UIKit
+
+class ViewTreesViewController: UIViewController {
+    var trees: [Tree] = [] {
+        didSet {
+            viewTreesCollectionview.reloadData()
+        }
+    }
+    var viewModel: ViewTreesViewModel? {
+        didSet {
+            viewModel?.viewDelegate = self
+            title = viewModel?.title
+        }
+    }
+    @IBOutlet weak var viewTreesCollectionview: UICollectionView!
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        viewTreesCollectionview.delegate = self
+        viewTreesCollectionview.dataSource = self
+//      collectionView.register(ViewTreesCollectionViewCell.self, forCellWithReuseIdentifier: "ViewTreesCollectionViewCell")
+        viewModel?.fetchTrees()
+    }
+}
+// MARK: - CollectionView cell implementation
+extension ViewTreesViewController: UICollectionViewDelegate, UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return trees.count
+    }
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "ViewTreesCollectionViewCell", for: indexPath) as? ViewTreesCollectionViewCell
+        let tree = trees[indexPath.item]
```

Comment:
So looks like we are fetching data on the main thread here. This will cause the UI to freeze up. We should load the image inside the cell, we can then also make use of cell reuse and cancel any ongoing download tasks. Otherwise we could end up with incorrect images on cells.

Also there is a lot of force unwrapping  - I think SwiftLint will throw a warning about this. We shouldn't force unwrap as it can cause a crash, we should handle the missing data more gracefully.

Happy to help you with this if you wish.

Reply:
I will fix that

Reply:
I made some changes in the code to fetch the pics instead of using main thread. However it didn't work out. You can fix this and please let me know.

Reply:
Here is a tutorial you can use to learn about this: https://cocoacasts.com/fm-3-download-an-image-from-a-url-in-swift

Reply:
you can help me on this.

Reply:
hope we have fixed this issue